### PR TITLE
planner: fix unique key get wrong row count when to query the NULL value | tidb-test=pr/2390

### LINKED ISSUE #56116: unique key get wrong row count when to query the Null value
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```
create table t2(id bigint(20) DEFAULT NULL, UNIQUE KEY index_on_id (id));
insert into t2 values (), (), ();
analyze table t2;
explain select count(*) from t2 where id is null;
```

### 2. What did you expect to see? (Required)

```
StreamAgg_17 1.00 root  funcs:count(Column#5)->Column#3,
└─IndexReader_18 1.00 root  index:StreamAgg_9,
  └─StreamAgg_9 1.00 cop[tikv]  funcs:count(1)->Column#5,
    └─IndexRangeScan_16 3.00 cop[tikv] table:t2, index:index_on_id(id) range:[NULL,NULL], keep order:false,
```
### 3. What did you see instead (Required)

```
StreamAgg_17 1.00 root  funcs:count(Column#5)->Column#3,
└─IndexReader_18 1.00 root  index:StreamAgg_9,
  └─StreamAgg_9 1.00 cop[tikv]  funcs:count(1)->Column#5,
    └─IndexRangeScan_16 1.00 cop[tikv] table:t2, index:index_on_id(id) range:[NULL,NULL], keep order:false,
```

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close #56116

Problem Summary:

### What changed and how does it work?

There will be issues with the row count under index/unique index. Previously, we would uniformly count the row count for this ```PointGet``` as ```1```, but if it is null, it will no longer be 1 and must be calculated according to the null count.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

From internal test cases

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: pkg/planner/cardinality/row_count_index.go [hunk 1]
```diff
@@ -251,10 +251,14 @@ func getIndexRowCountForStatsV2(sctx context.PlanContext, idx *statistics.Index,
...
-					if debugTrace {
-						debugTraceEndEstimateRange(sctx, 1, debugTraceUniquePoint)
+					if !indexRange.IsOnlyNull() {
+						totalCount++
+						if debugTrace {
+							debugTraceEndEstimateRange(sctx, 1, debugTraceUniquePoint)
+						}
+						continue
 					}
+					totalCount = float64(idx.NullCount)
```
Comment (winoros) [inline pkg/planner/cardinality/row_count_index.go:261]: Seems that we forget to add a `debugTraceEndEstimateRange`.

Comment (hawkingrei) [reply pkg/planner/cardinality/row_count_index.go:261]: Done

**Review by qw4990 — APPROVED** [review]

**Pull request merged**